### PR TITLE
Xsrust/481

### LINKED ISSUE #146: Create email notification system
Create email notification system
* new tab under Edit Profile for 'Notification Preferences' (see blue notepad in UX wireframes for functional specs)

UX wireframes @JEK-III : http://h90c3m.axshare.com/#g=1&p=profile

Pivotal ticket: https://www.pivotaltracker.com/story/show/119416081

dmponline_userprofile

DMPTool notifications - review/revise this list
dmptool_user_profile_notifcation_preferences

### PR DESCRIPTION
**Pull request by xsrust**
This addresses issue #481 Should also fix #146 as it changes the code in the controller and is no longer throwing errors when notification preferences are saved.

Implementation is detailed on the ticket.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -42,24 +42,8 @@ def filter(query)
...
-  #   -relies on protected_attributes gem as syntax depricated in rails 4.2
-  #accepts_nested_attributes_for :roles
-  #attr_accessible :password_confirmation, :encrypted_password, :remember_me,
-  #                :id, :email, :firstname, :last_login,:login_count, :orcid_id,
-  #                :password, :shibboleth_id, :user_status_id, :surname,
-  #                :user_type_id, :org_id, :skip_invitation, :other_organisation,
-  #                :accept_terms, :role_ids, :dmponline3, :api_token,
-  #                :organisation, :language, :language_id, :org, :perms,
-  #                :confirmed_at, :org_id
-
```
Comment (vyruss) [inline app/models/user.rb]: :+1:

File: app/models/user.rb [hunk 2]
```diff
@@ -223,17 +207,6 @@ def can_grant_api_to_orgs?
...
-  #
-  # @return [String] the organisation type
-=begin
-  def org_type
-    org_type = org.organisation_type
-    return org_type
-  end
-=end
-
-  ##
```
Comment (vyruss) [inline app/models/user.rb]: :+1:

File: config/routes.rb [hunk 3]
```diff
@@ -220,34 +223,6 @@
...
-#        get 'share'
-#        get 'export'
-#        post 'invite'
-#      end
-#      collection do
-#        get 'possible_templates'
-#        get 'possible_guidance'
-#      end
-#    end
-
```
Comment (vyruss) [inline config/routes.rb]: :+1:

**Comment by vyruss** [comment]
**This PR needs a `db:migrate` to be run for the new `prefs` table.**

**Comment by briri** [comment]
Looks good Sam.
thanks for clearing out the old commented code in routes and the user model. Looks good. That booleanize_hash method may be useful elsewhere. we can move it to the base controller or a mixin/concern later if we need to use it elsewhere.

**Comment by xsrust** [comment]
Thanks.  I didnt know if there were any other uses for it, but had a feeling that there might be some elsewhere in the code.

**Pull request merged**